### PR TITLE
Support .Net Standard

### PR DESCRIPTION
**Pull request by YairHalberstadt**
I have tested deploying this as a Nuget Package, referencing and using the nuget package from a .Net Standard class library, and then calling the library from both .Net Core and .Net Framework. Everything works as expected.

I hope that once this is merged, supporting .Net Standard for IronPython should be trivial.

### CODE REVIEW

File: Package/nuget/DynamicLanguageRuntime.nuspec [hunk 1]
```diff
@@ -24,6 +24,12 @@
         <dependency id="System.CodeDom" version="4.5.0" />
         <dependency id="System.Configuration.ConfigurationManager" version="4.5.0" />
       </group>
+      <group targetFramework="netstandard">
+        <dependency id="System.CodeDom" version="4.5.0" />
+        <dependency id="System.Configuration.ConfigurationManager" version="4.5.0" />
+        <dependency id="System.Reflection.Emit" version="4.6.0-preview.19073.11" />
```
Comment (YairHalberstadt) [inline Package/nuget/DynamicLanguageRuntime.nuspec]: I am using the prerelease versions as I believe (but am not sure) that the prerelease versions will throw a PlatformNotSupportedException on AOT platforms, whereas the stable versions wont. I hope to give that a test at some point.

Comment (YairHalberstadt) [reply Package/nuget/DynamicLanguageRuntime.nuspec]: Tested, and I am correct. Stable will throw this exception:

> System.Exception: 'Method 'TypeBuilder.DefineConstructor(MethodAttributes, CallingConventions, Type[])' was not included in compilation, but was referenced in DelegateUtils.EmitCallSiteDelegateType(int). There may have been a missing assembly.'

The preview will throw this, more helpful exception:

> System.PlatformNotSupportedException: PlatformNotSupported_RefEmit. For more information, visit http://go.microsoft.com/fwlink/?LinkId=623485

File: make.ps1 [hunk 2]
```diff
@@ -4,7 +4,7 @@ Param(
     [Parameter(Position=1)]
     [String] $target = "release",
     [String] $configuration = "Release",
-    [String[]] $frameworks=@('net45','netcoreapp2.0','netcoreapp2.1'),
+    [String[]] $frameworks=@('net45','netcoreapp2.0','netcoreapp2.1', 'netstandard2.0'),
```
Comment (slozier) [inline make.ps1]: This list is only used for testing and they're not being build against `netstandard2.0` in this PR so we shouldn't include this.

File: Src/Microsoft.Dynamic/Generation/ILGen.cs [hunk 3]
```diff
@@ -251,7 +251,7 @@ public class ILGen {
             _ilg.EmitCall(opcode, methodInfo, optionalParameterTypes);
         }
 
-#if !NETCOREAPP2_0 || NETCOREAPP2_1
+#if (!NETCOREAPP2_0 && !NETSTANDARD2_0) || NETCOREAPP2_1
```
Comment (slozier) [inline Src/Microsoft.Dynamic/Generation/ILGen.cs]: Nit: I'm not sure why we had the `|| NETCOREAPP2_1` here. Doesn't do anything and can be removed.

File: make.ps1 (hunk 2 above)
Comment (YairHalberstadt) [reply make.ps1]: Done

File: Src/Microsoft.Dynamic/Generation/ILGen.cs (hunk 3 above)
Comment (YairHalberstadt) [reply Src/Microsoft.Dynamic/Generation/ILGen.cs]: Done

**Comment by slozier** [comment]
Thanks again!

**Pull request merged**